internal/state/state.go

```diff
@@ -66,8 +67,9 @@ func (c *state) newClient() hcapi2.Client {
 		if filePath := config.OptionDebugFile.Get(c.config); filePath == "" {
 			opts = append(opts, hcloud.WithDebugWriter(os.Stderr))
 		} else {
-			writer, _ := os.Create(filePath)
-			opts = append(opts, hcloud.WithDebugWriter(writer))
+			f, _ := os.OpenFile(filePath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+			_, _ = f.WriteString("--- Command:\n" + strings.Join(os.Args, " ") + "\n\n\n\n")
```

Comment:
Since we are adding this, do you think it's worth quoting the args to detect quoting issues?